### PR TITLE
add two short test videos, use them in component view test

### PR DESCRIPTION
**Pull request by orm011**
speeds component view test substantially.

### CODE REVIEW

File: pixeltable/tests/utils.py [hunk 1]
```diff
@@ -221,6 +221,11 @@ def get_video_files(include_bad_video=False) -> List[str]:
         glob_result = [f for f in glob_result if 'bad_video' not in f]
     return glob_result
 
+def get_simple_video_files() -> List[str]:
```
Comment (mkornacker) [inline pixeltable/tests/utils.py]: What does simple mean?

Are these easier to decode or do they have fewer frames? If the latter, that might be problematic, because it might test fewer corner cases in the code.

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/mkornacker/pixeltable/73)* status: 0 of 4 files reviewed, 1 unresolved discussion (waiting on @mkornacker)

___
*[`pixeltable/tests/utils.py` line 224 at r1](https://reviewable.io/reviews/mkornacker/pixeltable/73#gh-1468286510:-Np7tSJq-dTQOgmJrY-D:bl45h4k) ([raw file](https://github.com/mkornacker/pixeltable/blob/54cd2e57ab3637166ed685ae3a06539884c0fb91/pixeltable/tests/utils.py#L224)):*
<details><summary><i>Previously, mkornacker (Marcel Kornacker) wrote…</i></summary><blockquote>

What does simple mean?

Are these easier to decode or do they have fewer frames? If the latter, that might be problematic, because it might test fewer corner cases in the code.
</blockquote></details>

I think a big factor in decoding cost is the frame size.  These are roughly 224x224. The two are of different size. 

They are simple in content ( frame number on plain background ).

One is 17 frames, the other 13. Not sure if there is some number that you specifically want based on some knowledge of the code. 

 ( otherwise why not make them even longer than before )

**Comment by orm011** [comment]
I have now made a video that's the same length as the longest video previously used (462 frames) but with much smaller resolution. It's enough to make things go much faster. (all numbers are on a apple silicon v2 machine, so this is not about bad harware)

before:
44.42s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot1

44.29s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot2

36.39s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot3

36.01s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot4

32.15s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_chained\_views

4.67s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_add\_column

after:
slowest 10 durations ===============================================================================================

1.39s setup    pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_basic

0.95s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot4

0.94s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot2

0.92s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot1

0.87s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot3

0.81s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_chained\_views

0.16s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_basic

0.11s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_add\_column

0.07s call     pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_update

0.03s setup    pixeltable/pixeltable/tests/test\_component\_view.py::TestComponentView::test\_snapshot1

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/73)* status: 0 of 5 files reviewed, 1 unresolved discussion (waiting on @mkornacker)

___
*[`pixeltable/tests/utils.py` line 224 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/73#gh-1468286510:-NtrHcY_5W90jp6TfVDR:bw6f20c) ([raw file](https://github.com/pixeltable/pixeltable/blob/54cd2e57ab3637166ed685ae3a06539884c0fb91/pixeltable/tests/utils.py#L224)):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

I think a big factor in decoding cost is the frame size.  These are roughly 224x224. The two are of different size. 

They are simple in content ( frame number on plain background ).

One is 17 frames, the other 13. Not sure if there is some number that you specifically want based on some knowledge of the code. 

 ( otherwise why not make them even longer than before )
</blockquote></details>

I have now updated this to use a long video and a short video. The long video is as long as before. But the video is simpler. It's still much faster. Does that sound good?

**Comment by orm011** [comment]
*[`pixeltable/tests/utils.py` line 224 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/73#gh-1468286510:-NtrIbKF0FsfF1G7FPaO:b1sgfnl) ([raw file](https://github.com/pixeltable/pixeltable/blob/54cd2e57ab3637166ed685ae3a06539884c0fb91/pixeltable/tests/utils.py#L224)):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

I have now updated this to use a long video and a short video. The long video is as long as before. But the video is simpler. It's still much faster. Does that sound good?
</blockquote></details>

(simpler = low resolution frames)

**Review by aaron-siegel — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/73)* status: 0 of 5 files reviewed, 2 unresolved discussions (waiting on @mkornacker and @orm011)

___
*[`pixeltable/tests/utils.py` line 224 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/73#gh-1468286510:-NtrO_y-5n62AD0tsVks:bf41wpy) ([raw file](https://github.com/pixeltable/pixeltable/blob/54cd2e57ab3637166ed685ae3a06539884c0fb91/pixeltable/tests/utils.py#L224)):*
<details><summary><i>Previously, orm011 (Oscar Moll) wrote…</i></summary><blockquote>

(simpler = low resolution frames)
</blockquote></details>

How about `get_test_video_files`?
___
*[`pixeltable/tests/utils.py` line 292 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/73#-NtrOl194nRTfe64m9s-:-NtrOl194nRTfe64m9s0:brrx6t3) ([raw file](https://github.com/pixeltable/pixeltable/blob/9305d77058732ae594513570663b4069143cd5aa/pixeltable/tests/utils.py#L292)):*
> ```Python
>         pytest.skip(f'Package `{package}` is not installed.')
> 
> def make_test_video(
> ```

Is the intent to run this once (or occasionally when something changes) as a standalone, rather than as part of the normal user/dev workflow?

If so, let's move it to the pixeltable.tool package (which is meant for standalone devops/admin/etc tools).

Also include an example in the comments to demonstrate the intended usage.

At some point (not necessarily as part of this PR) we should think about how to test these tools. (`create_test_db_dump` has the same concern.) I can almost guarantee that if we don't include them in our testing, then at some point they will suffer bitrot.

**Review by orm011 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/73)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @aaron-siegel and @mkornacker)

___
*[`pixeltable/tests/utils.py` line 224 at r1](https://reviewable.io/reviews/pixeltable/pixeltable/73#gh-1468286510:-NtrSYoq0kbcBl2RMw1K:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/54cd2e57ab3637166ed685ae3a06539884c0fb91/pixeltable/tests/utils.py#L224)):*
<details><summary><i>Previously, aaron-siegel (Aaron Siegel) wrote…</i></summary><blockquote>

How about `get_test_video_files`?
</blockquote></details>

Done.
___
*[`pixeltable/tests/utils.py` line 292 at r2](https://reviewable.io/reviews/pixeltable/pixeltable/73#-NtrOl194nRTfe64m9s-:-NtrUZIy9lDn0NU56_tZ:b-896fix) ([raw file](https://github.com/pixeltable/pixeltable/blob/9305d77058732ae594513570663b4069143cd5aa/pixeltable/tests/utils.py#L292)):*
<details><summary><i>Previously, aaron-siegel (Aaron Siegel) wrote…</i></summary><blockquote>

Is the intent to run this once (or occasionally when something changes) as a standalone, rather than as part of the normal user/dev workflow?

If so, let's move it to the pixeltable.tool package (which is meant for standalone devops/admin/etc tools).

Also include an example in the comments to demonstrate the intended usage.

At some point (not necessarily as part of this PR) we should think about how to test these tools. (`create_test_db_dump` has the same concern.) I can almost guarantee that if we don't include them in our testing, then at some point they will suffer bitrot.
</blockquote></details>

Done.

**Comment by mkornacker** [comment]
I would like to take a completely different approach to solving this problem. Let's discuss.

**Review by orm011 — COMMENTED** [review]
I pulled your changes based on replacing FrameIterator,  kept them, but also applied this PR on top of them, there are still benefits 161.25 secs. vs  186.93 secs (this is just from TestComponentView, the part still using frame iterator), 

I'd propose we still take this PR,  which includes long video so your concern about not reaching the same edge cases was addressed (let me know if thats not the concern). 

Additionally, the new tool included to create new videos will be useful so that we can make custom test videos rather than download large videos.

PS. The bulk of the remaining test latency is heavily affected by TestVideo, which we can easily reduce by merely rescaling the test videos (keeping them the same length). I think we should do that, its a simple change.

20.30s call     pixeltable/pixeltable/tests/test\_video.py::TestVideo::test\_basic
14.80s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_create\_video\_table
13.24s call     pixeltable/pixeltable/tests/test\_document.py::TestDocument::test\_doc\_splitter
12.57s call     pixeltable/pixeltable/tests/test\_video.py::TestVideo::test\_make\_video
10.83s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_validate\_external\_url
10.21s call     pixeltable/pixeltable/tests/test\_document.py::TestDocument::test\_doc\_splitter\_headings
8.54s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_create\_s3\_image\_table
6.09s call     pixeltable/pixeltable/tests/test\_functions.py::TestFunctions::test\_gpt\_4\_vision
4.68s call     pixeltable/pixeltable/tests/test\_video.py::TestVideo::test\_query
4.57s call     pixeltable/pixeltable/tests/test\_video.py::TestVideo::test\_computed\_cols
   

> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/73#-:-NtxqpVDA2pLRcveKE1M:b-ykchdj)* status: 0 of 6 files reviewed, 2 unresolved discussions (waiting on @aaron-siegel and @mkornacker)

**Review by orm011 — COMMENTED** [review]
I just pushed an update to this PR where I also add rescaled versions of the videos currently in data/video, and use those by default.
There's some cleanup to do in how this is done. but here's the benefit:

 usr time   79.05 secs    0.24 millis   79.05 secs

13.61s call     pixeltable/pixeltable/tests/test\_document.py::TestDocument::test\_doc\_splitter
10.19s call     pixeltable/pixeltable/tests/test\_document.py::TestDocument::test\_doc\_splitter\_headings
7.77s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_create\_video\_table
6.76s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_video\_url
5.35s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_validate\_external\_url
4.21s call     pixeltable/pixeltable/tests/test\_functions.py::TestFunctions::test\_gpt\_4\_vision
3.98s call     pixeltable/pixeltable/tests/test\_video.py::TestVideo::test\_basic
3.72s call     pixeltable/pixeltable/tests/test\_functions.py::TestFunctions::test\_sentence\_transformer
3.25s call     pixeltable/pixeltable/tests/test\_table.py::TestTable::test\_create\_s3\_image\_table
2.97s call     pixeltable/pixeltable/tests/test\_functions.py::TestFunctions::test\_openai

We need to be careful of tests exploding in duration, we write once but run them all the time.

> *[Reviewable](https://reviewable.io/reviews/pixeltable/pixeltable/73#-:-NtzuAvb0E-4_dF03le5:b-z793k3)* status: 0 of 9 files reviewed, 2 unresolved discussions (waiting on @aaron-siegel and @mkornacker)

**Review by aaron-siegel — APPROVED** [review]
